gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/ResolveProductDependenciesTask.java

```diff
@@ -0,0 +1,347 @@
+                                return Stream.empty();
+                            }
+                            manifest = new Manifest(zipFile.getInputStream(manifestEntry));
+                        } catch (IOException e) {
+                            log.warn(
+                                    "IOException encountered when processing artifact '{}', file '{}', {}",
+                                    artifactName,
+                                    artifact.getFile(),
+                                    e);
+                            return Stream.empty();
+                        }
+
+                        pdeps = Optional.ofNullable(manifest.getMainAttributes()
+                                .getValue(RecommendedProductDependencies.SLS_RECOMMENDED_PRODUCT_DEPS_KEY));
+                    }
+                    if (!pdeps.isPresent()) {
+                        log.debug(
+                                "No product dependency found for artifact '{}', file '{}'",
+                                artifactName,
+                                artifact.getFile());
+                        return Stream.empty();
+                    }
+
+                    try {
+                        RecommendedProductDependencies recommendedDeps = CreateManifestTask.jsonMapper.readValue(
+                                pdeps.get(), RecommendedProductDependencies.class);
+                        return recommendedDeps.recommendedProductDependencies().stream()
+                                .peek(productDependency -> log.info(
+                                        "Product dependency recommendation made by artifact '{}', file '{}', "
+                                                + "dependency recommendation '{}'",
+                                        artifactName,
+                                        artifact,
+                                        productDependency))
+                                .map(recommendedDep -> new ProductDependency(
+                                        recommendedDep.getProductGroup(),
+                                        recommendedDep.getProductName(),
+                                        recommendedDep.getMinimumVersion(),
+                                        recommendedDep.getMaximumVersion(),
+                                        recommendedDep.getRecommendedVersion(),
+                                        recommendedDep.getOptional()));
+                    } catch (IOException | IllegalArgumentException e) {
+                        log.debug(
+                                "Failed to load product dependency for artifact '{}', file '{}', '{}'",
+                                artifactName,
+                                artifact,
+                                e);
+                        return Stream.empty();
+                    }
+                })
+                .filter(this::isNotSelfProductDependency)
```

Comment:
we should probably do the filtering inside of createManifest task. It will also allow us to remove a few inputs into this task

Reply:
eh - we still need the serviceName and serviceGroup parameters in the resolving task because we check and thow error if declare self reference explicitly.